todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/tasks/TasksViewModel.kt

```diff
@@ -84,49 +108,50 @@ class TasksViewModel(
 
     /**
      * Sets the current task filtering type.
-
+     *
      * @param requestType Can be [TasksFilterType.ALL_TASKS],
-     * *                    [TasksFilterType.COMPLETED_TASKS], or
-     * *                    [TasksFilterType.ACTIVE_TASKS]
+     * [TasksFilterType.COMPLETED_TASKS], or
+     * [TasksFilterType.ACTIVE_TASKS]
      */
-    fun updateFiltering() {
+    fun setFiltering(requestType: TasksFilterType) {
+        _currentFiltering = requestType
+
         // Depending on the filter type, set the filtering label, icon drawables, etc.
-        when (currentFiltering) {
+        when (requestType) {
             TasksFilterType.ALL_TASKS -> {
-                setFilter(R.string.label_all, R.string.no_tasks_all,
-                        R.drawable.ic_assignment_turned_in_24dp, true)
+                _currentFilteringLabel.value = context.getString(R.string.label_all)
```

Comment:
I kinda' like the method used before (maybe with a better name). why was it removed? can we bring it back?

Reply:
Bad merge, done